### ISSUE TITLE
ReplayFile.readAsLines strips a trailing blank line

### ISSUE THREAD

**Issue by jamesderlin** [google/file.dart#146]
While adding tests for https://github.com/google/file.dart/issues/142, I discovered that `ReplayFile` has the opposite problem: it discards trailing blank lines.

```dart
          f.writeAsStringSync('\n');
          expect(f.readAsLinesSync(), <String>['']); // Fails; actual value is [].

          f.writeAsStringSync('\n\n');
          expect(f.readAsLinesSync(), <String>['', '']); // Fails; actual value is ['']
```

I *think* that this is happening when the blob is written (when `ReplayFile.readAsLinesSync` is called for the first case, the `blobToBytes` converter returns an empty `Uint8List`.)

**Comment by jamesderlin** [comment]
Oh, the reason why it has the opposite problem is *because* of my fix for #142; the `ReplayFile` tests delegate to `MemoryFile`.  I'm still not sure exactly why it's broken; I'm wondering if it's due to the fact that `readAsLines` is lossy and can't distinguish between a file with a terminating newline and a file without.

**Comment by jamesderlin** [comment]
I understand what's going on now.

In tests, `RecordingFile.readAsLines` delegates to `MemoryFile.readAsLines` and tries to save the returned `List<String>` by flattening it back into a `String` via `List.join('\n')`.  Because of my fix for #142, the flattening process saves the file without a terminating newline even if one was originally present.

This can be fixed by making `RecordingFile.readAsLines`/`readAsLinesSync` *always* write a terminating newline.  Of course, this creates the opposite problem where `RecordingFile.readAsLines` adds a terminating newline to the recorded file when there wasn't one originally.  And indeed, such a change causes `recording_test.dart`'s `readAsLines` tests to fail when the file content doesn't have a terminating newline.

*However*, I claim that it's the more correct thing to do. `readAsLines` makes sense only on text files, and the POSIX definition of a "text file" requires a terminating newline.  I therefore claim that the file content in the affected tests should have terminating newlines in the first place.
